src/main/java/io/stargate/sgv2/jsonapi/service/shredding/tables/RowShredder.java

```diff
@@ -91,11 +91,24 @@ public WriteableTableRow shred(TableSchemaObject table, JsonNode document) {
    */
   public static Object shredValue(JsonNode value) {
     return switch (value.getNodeType()) {
-      case NUMBER -> value.decimalValue();
+      case NUMBER -> shredNumber(value);
       case STRING -> value.textValue();
       case BOOLEAN -> value.booleanValue();
       case NULL -> null;
       default -> throw new RuntimeException("Unsupported type");
     };
   }
+
```

Comment:
I know this class is not finished / high quality - can you add a quick comment to say the different types for numbers need to match the codecs in JSONCodecRegistry

Reply:
Added something in class javadoc, but will also add it right here too.